src/main/java/com/linagora/james/mailets/ExtractPhoneNumberMailet.java

```diff
@@ -0,0 +1,151 @@
+ * GNU Affero General Public License for more details.                         *
+ *                                                                             *
+ * You should have received a copy of the GNU Affero General Public License    *
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.       *
+ *******************************************************************************/
+
+package com.linagora.james.mailets;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableList;
+import com.google.i18n.phonenumbers.PhoneNumberMatch;
+import com.google.i18n.phonenumbers.PhoneNumberUtil;
+import com.linagora.james.mailets.json.UUIDGenerator;
+import org.apache.james.mime4j.stream.MimeConfig;
+import org.apache.james.util.mime.MessageContentExtractor;
+import org.apache.mailet.Mail;
+import org.apache.mailet.MailetException;
+import org.apache.mailet.base.GenericMailet;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.mail.MessagingException;
+import javax.swing.text.html.Option;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * This mailet adds a header to the mail if it found a phone number in the signature of the mail
+ *
+ * <pre>
+ * <code>
+ * &lt;mailet match="All" class="ExtractPhoneNumberMailet"&gt;
+ *    &lt;headerName&gt; <i>The classification message header name, default=X-Classification-Guess</i> &lt;/headerName&gt;
+ * &lt;/mailet&gt;
+ * </code>
+ * </pre>
+ *
+ * Sample Configuration:
+ *
+ * <pre>
+ * <code>
+ * &lt;mailet match="All" class="ExtractPhoneNumberMailet"&gt;
+ *    &lt;headerName&gt;X-Phone-Number&lt;/headerName&gt;
+ * &lt;/mailet&gt;
+ * </code>
+ * </pre>
+ *
+ */
+public class ExtractPhoneNumberMailet extends GenericMailet {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ExtractPhoneNumberMailet.class);
+
+    static final String HEADER_NAME = "headerName";
+    static final String HEADER_NAME_DEFAULT_VALUE = "X-Phone-Number";
+
+    @VisibleForTesting String headerName;
+    private final UUIDGenerator uuidGenerator;
+    private final PhoneNumberUtil phoneNumberUtil;
+
+    private static final MimeConfig MIME_ENTITY_CONFIG = MimeConfig.custom()
+        .setMaxContentLen(-1)
+        .setMaxHeaderCount(-1)
+        .setMaxHeaderLen(-1)
+        .setMaxHeaderCount(-1)
+        .setMaxLineLen(-1)
+        .build();
+
+    public ExtractPhoneNumberMailet() {
+        this(new UUIDGenerator());
+    }
+
+    @VisibleForTesting
+    ExtractPhoneNumberMailet(UUIDGenerator uuidGenerator) {
+        this.uuidGenerator = uuidGenerator;
+        this.phoneNumberUtil = PhoneNumberUtil.getInstance();
+    }
+
+    @Override
+    public void init() throws MessagingException {
+        headerName = getInitParameter(HEADER_NAME, HEADER_NAME_DEFAULT_VALUE);
+        if (LOGGER.isDebugEnabled()) {
```

Comment:
actually, when using the placeholders '{}' and not in debug, cost is zero, the code is removed completely